src/chain.js

```diff
@@ -11,19 +11,25 @@ var map = require('./map');
  *
  * Dispatches to the `chain` method of the second argument, if present,
  * according to the [FantasyLand Chain spec](https://github.com/fantasyland/fantasy-land#chain).
+ * Treats the second argument as a FantasyLand Chain, when the the second argument is a function
+ * of the form `a -> x -> b`.
```

Comment:
Remember we are only doing *half* of Haskell. 

> I don't see how to get from Chain m => (a -> m b) -> m a -> m b to the signature you're proposing. Do you?

Yes, it's odd alright; clearly `head` is `x -> a` and `append` is `a -> x -> b`, and the other signature makes more sense. But I'm not proposing it. It's already a fact today, since https://github.com/ramda/ramda/commit/35acea0dd9a32f0327737689ac9cc7b279d8d058

I am simply recommending we document it. If you are suggesting we should treat this behavior as a defect, that is a very different conversation.